### ISSUE TITLE
Chapter 2 - Feature_importances versus the pipeline

### ISSUE THREAD

**Issue by Rosseel** [ageron/handson-ml2#191]
When making a select and predict pipeline, the feature importance's are supposedly calculated beforehand.
But how can it do so considering we create new features from scratch in the pipeline itself?
Or is the feature_importances var updated after the creation of said new features?

prepare_select_and_predict_pipeline = Pipeline([
    ('preparation', full_pipeline),
    ('feature_selection', TopFeatureSelector(feature_importances, k)),
    ('svm_reg', SVR(**rnd_search.best_params_))
])

**Comment by ageron** [comment]
Thanks for your question. Indeed, in this example we assume that the feature importances were computed ahead of time: in other words, we ran the preparation pipeline once already, we ran some analysis and we figured out which features were most important. Then the purpose of this new pipeline is to add the feature selection step which keeps only the top k features. Since we don't know what the best value of k is, this new pipeline could be used to try different values of k and find which one gives the best results on the validation set.

It would probably be better to compute the feature importances as part of the pipeline itself (between the preparation step and the feature selection step).

Hope this helps.
